client/my-sites/customer-home/main.jsx

```diff
@@ -83,10 +87,10 @@ const Home = ( {
 	}, [ detectedCountryCode, sitePlan, isNew7DUser ] );
 
 	useEffect( () => {
-		if ( getQueryArgs().celebrateLaunch === 'true' && isSuccess ) {
+		if ( getQueryArgs().celebrateLaunch === 'true' && ! isFetching && isSuccess ) {
```

Comment:
This makes it seem like `isSuccess` is getting prematurely marked as `true` ?  Did you find that this was being set before `isFetching` had completed? Or is it a situation with a subsequent request?

Reply:
> Did you find that this was being set before isFetching had completed? Or is it a situation with a subsequent request?

My hypothesis is that, if a value has been cached, `isSuccess` will always return true. I'm wondering if, in production, somehow it thinks that the request is cached?

We use a cache key of site id, which should be unique for each request, but there might be a period of time, during data hydration, where site id is `null`. Because we rely on `isSuccess` for the celebration modal, I then wonder if that's why we're loading the celebration modal prematurely.